posthog/queries/query_date_range.py

```diff
@@ -18,17 +23,110 @@ def __init__(self, filter, team: Team, should_round: Optional[bool] = None, tabl
         self._should_round = should_round
 
     @cached_property
-    def date_to_param(self) -> Optional[str]:
-        return self._date_param(
-            self._filter.date_to, self._filter._date_to and not self._filter.date_to_has_explicit_time
-        )
+    def date_to_param(self) -> datetime:
+
+        if not self._filter._date_to and self._filter.interval == "hour":
+            return self._localize_to_team(self._now) + relativedelta(minutes=1)
+
+        date_to = self._now
+        if isinstance(self._filter._date_to, str):
+            date_to = self._parse_date(self._filter._date_to)
+        elif isinstance(self._filter._date_to, datetime):
+            date_to = self._localize_to_team(self._filter._date_to)
+
+        return date_to
+
+    def get_earliest_timestamp(self):
+        try:
+            earliest_date = get_earliest_timestamp(self._team.pk)
+        except IndexError:
+            return timezone.now()  # TODO: fix
+        else:
+            return earliest_date
+
+    @cached_property
+    def date_from_param(self) -> datetime:
+        date_from: datetime
+        if self._filter._date_from == "all":
+            date_from = self.get_earliest_timestamp()
+        elif isinstance(self._filter._date_from, str):
+            date_from = self._parse_date(self._filter._date_from)
+        elif isinstance(self._filter._date_from, datetime):
+            date_from = self._localize_to_team(self._filter._date_from)
+        else:
+            date_from = self._localize_to_team(
+                timezone.now().replace(hour=0, minute=0, second=0, microsecond=0)
+                - relativedelta(days=DEFAULT_DATE_FROM_DAYS)
+            )
+
+        return date_from
 
-    def _date_param(self, target_date, should_convert) -> str:
-        return format_ch_timestamp(
-            target_date,
-            convert_to_timezone=self._team.timezone if should_convert else None,
+    @cached_property
+    def _now(self):
+        return self._localize_to_team(timezone.now())
+
+    def _localize_to_team(self, target: datetime):
+        return (
+            target.astimezone(pytz.timezone(self._team.timezone))
+            if target.tzinfo is None
+            else target.astimezone(pytz.timezone(self._team.timezone))
         )
 
+    def _parse_date(self, input):
```

Comment:
This whole class feels complicated enough that I think it would be nice to have unit tests around this behaviour? Hard to see exactly what's changing using just the snapshots, and I'm pretty sure coverage here right now is very small.